src/routes/event/details.tsx

```diff
@@ -86,23 +83,32 @@ export default function EventDetails() {
           label="Event category"
           defaultValue={eventDetails?.eventCategory}
         />
-        <Stack direction="row" spacing={2}>
+        <Stack direction="row" spacing={2} width="100%">
           {auth.currentUser &&
             auth.currentUser.uid === eventDetails?.eventOrganizer && (
-              <Button variant="contained" color="primary">
+              <Button variant="contained" color="primary" fullWidth>
                 Edit
               </Button>
             )}
           <Button
             variant="contained"
-            color="primary"
+            color="secondary"
+            fullWidth
             onClick={() => {
               navigate("/");
             }}
           >
             Go Back
           </Button>
         </Stack>
+        <Stack direction="row" spacing={2} width="48%">
+          {auth.currentUser &&
+            auth.currentUser.uid === eventDetails?.eventOrganizer && (
+              <Button variant="contained" color="error" fullWidth>
+                Delete
+              </Button>
+            )}
+        </Stack>
```

Comment:
This seems to be repeating, maybe think of a wrapper that can handle this, that you can pass it a prop for the type of a user. Just a thought